### PR TITLE
Don't trim long comments as they can contain HTML, but instead overfl…

### LINKED ISSUE #10183: Comments not visible or partially visible
**Describe the bug**
Comments not visible or partially visible

**To Reproduce 1-2**
Steps to reproduce the behaviour:
1. Add comment with a URL with many characters (250+)
2. Comment section is empty

**To Reproduce 2-2**
Steps to reproduce the behaviour:
1. Add comment with multiple short URL's 
2. Comment section only shows a few URL's

**Expected behaviour**
See comment and URL

**Your environment (please complete the following information):**
 - Browser: Chrome
 - Quality-time version: v5.17.1

### PR DESCRIPTION
**Pull request by fniessink**
…ow the table cell.

Fixes #10183.

### CODE REVIEW

File: components/frontend/src/widgets/DivWithHTML.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+export function DivWithHTML({ children }) {
+    const minHeight = 15
+    const maxHeight = 60
+    const ref = createRef()
+    // Keep track of whether the div is overflown, so the resize handle can be hidden when not needed
+    const [overflown, setOverflown] = useState(false)
+    const [scrollHeight, setScrollHeight] = useState(minHeight)
+    useLayoutEffect(() => {
+        setOverflown(ref.current.clientHeight < ref.current.scrollHeight)
+        setScrollHeight(ref.current.scrollHeight)
```
Comment (wkoot) [inline components/frontend/src/widgets/DivWithHTML.js:14]: I don't understand how this works to adjust the `scrollHeight`, but seems like some nice js magic :)

File: components/frontend/src/widgets/DivWithHTML.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    }, [ref])
+    return (
+        <div
+            ref={ref}
+            style={{
+                wordBreak: "break-word",
+                overflow: "auto",
+                minHeight: minHeight,
+                height: Math.min(maxHeight, scrollHeight),
+                resize: overflown ? "vertical" : "",
```
Comment (wkoot) [inline components/frontend/src/widgets/DivWithHTML.js:24]: So this means that the div can only grow vertically, i.e. in height, and will therefore not break table columns that it's in?

File: components/frontend/src/widgets/DivWithHTML.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+            }}
+            dangerouslySetInnerHTML={{ __html: children }}
```
Comment (wkoot) [inline components/frontend/src/widgets/DivWithHTML.js:26]: This inner html cannot break the parent bounding box and resizing rules (e.g. with something like display block)?

**Review by wkoot — APPROVED** [review]

File: components/frontend/src/widgets/DivWithHTML.js (hunk 1 above)
Comment (fniessink) [reply components/frontend/src/widgets/DivWithHTML.js:14]: The `ref.current.scrollHeight` isn't available until after the size of the children prop has been calculated. So we can't pass it directly to the div style. The `useLayoutEffect` hook runs after the children prop has been rendered and the scrollHeight is known.

File: components/frontend/src/widgets/DivWithHTML.js (hunk 2 above)
Comment (fniessink) [reply components/frontend/src/widgets/DivWithHTML.js:24]: Yes!

File: components/frontend/src/widgets/DivWithHTML.js (hunk 3 above)
Comment (fniessink) [reply components/frontend/src/widgets/DivWithHTML.js:26]: I don't think so, but if users want to shoot themselves in the foot that way they can try :-)

**Pull request merged**